R/datalog.R

```diff
@@ -10,13 +10,16 @@
 #'
 process_question_mark <- function(s) {
     ret <- gsub("`", "", s)
-    if(length(grep("?", ret)) > 0) {
+    if(length(grep("\\?", ret)) > 0) {
         regex <- "\\?\\(([^\\(\\)]*)\\)"
 
-        m <- stringr::str_match(ret, regex)
-        s <- sprintf("?%s", m[1, 2])
+        m <- stringr::str_extract_all(ret, regex)
+        s <- sapply(m[[1]], function(x){
+            sprintf("?%s", stringr::str_match(x,regex)[,2])})
         s <- gsub(" - ", "-", s)
-        ret <- stringr::str_replace(ret, regex, s)
+        # Fix named vector names to be regex safe:
+        names(s) <- sapply(names(s), function(x){stringr::str_replace_all(x, "(\\W)", "\\\\\\1")})
```

Comment:
What is this line for?

Reply:
This `\\` out special characters to allow `stringr::str_replace_all` to not get tripped up by `(` or `?`. 
`"?(a)" "?(b)" "?(c)"` goes to `"\\?\\(a\\)" "\\?\\(b\\)" "\\?\\(c\\)" `.

replace_all searches for `\\?\\(a\\)` in `s` and then turns it into the results from `ret`, in this case `?a`